src/app/trade-creator-page/trade-creator-preview/trade-creator-preview.component.ts

```diff
@@ -123,6 +125,13 @@ export class TradeCreatorPreviewComponent implements OnInit {
 
           // This will update the user's balance.
           this.appData.updateEverything(response.TxnHashHex, this._creatorCoinSuccess, this._creatorCoinFailure, this);
+
+          if (
+            this.creatorCoinTrade.followCreator &&
+            !this.followService._isLoggedInUserFollowing(this.creatorCoinTrade.creatorProfile.PublicKeyBase58Check)
+          ) {
+            this.followService._toggleFollow(true, this.creatorCoinTrade.creatorProfile.PublicKeyBase58Check);
+          }
```

Comment:
Okay so i think we need to call updateEverything after the follow transaction is completed as well or at least update the logged in user to show that the user is following the public key. If I buy a creator coin and have the Follow checkbox checked, I get my creator coins and the follow transaction goes through, but if I go to buy more of the creator coin, I still see the checkbox.

Reply:
I thought `_toggleFollow` updated the local cache of who you are following

Reply:
you're right - it does do that. I think something strange is happening since we're calling `updateEverything` around the same time as toggle follow. Sometimes after I buy a creator with the checkbox checked, I'll still have the option to follow. However, if I perform a refresh, it doesn't show up. My guess is that we've got some race condition where we get undesirable behavior when updateEverything or the follow transaction gets processed in a slightly different order.

Reply:
my guess is that we should call updateEverything once after toggling the follow instead of at the same time. In the event that the checkbox is not checked or the user already follows the creator, we can just call updateEverything